services/wallet/router/router_v2.go

```diff
@@ -326,6 +328,34 @@ func validateInputData(input *RouteInputParams) error {
 		}
 	}
 
+	if input.SendType == Swap {
+		if input.ToTokenID == "" {
+			return errors.New("toTokenID is required for Swap")
+		}
+		if input.TokenID == input.ToTokenID {
+			return errors.New("tokenID and toTokenID must be different")
+		}
+
+		// we can do this check, cause AmountIn is required in `RouteInputParams`
```

Comment:
the paraswap client allows setting both (input amount and min expected out amount) instead of using slippage, do we want to restrict that use case?

Reply:
> the paraswap client allows setting both (input amount and min expected out amount) instead of using slippage

Could you show me which one is the related param in this page https://developers.paraswap.network/api/get-rate-for-a-token-pair ?

When we ask API for estimation I don't see that we have slippage or src/dest amount, just an amount and side (sell/buy).
Later when user wants to send the swap tx, then slippage and src/dest amounts come into play.


> do we want to restrict that use case?

`AmountIn` is set as "required" param in `RouteInputParams` type and I wouldn't like to change that because of other processor types, while `AmountOut` is an optional param.

In the case of swap, you can provide:
- `AmountIn` or
- `AmountIn` set to 0 and `AmountOut`

that's why the validation in case of swap looks like it looks.

Reply:
ohhh sorry! Mixing the Tx sending and the Router part!
Yeah, slippage is not involved in the router flow, my bad.